src/ptychography/reconstruction/ssb/udf.py

```diff
@@ -1,11 +1,66 @@
 import numpy as np
+import numba
 
 from libertem.udf import UDF
 from libertem.common.container import MaskContainer
 
 from ptychography.reconstruction.ssb.trotters import generate_masks
 
 
+@numba.njit(fastmath=True, cache=True, parallel=True)
+def rmatmul_csc_fourier(n_threads, left_dense, right_data, right_indices, right_indptr,
+                        coordinates, row_exp, col_exp, res_inout):
+    '''
+    Fold :meth:`SSB_UDF.merge_dot_result` into a sparse dot product from
+    :meth:`libertem.common.numba._rmatmul_csc`
+
+    The result can be directly merged into the result buffer instead of
+    instantiating the intermediate dot result, which can be large. That way,
+    this method can process entire memory-mapped partitions efficiently.
+    Furthermore, it allows early skipping of trotters that are empty in this tile.
```

Comment:
Getting rid of that intermediate result is **so nice**, this is great!

Reply:
😁 thx.